### ISSUE TITLE
Incorrect recognition of the phone number

### ISSUE THREAD

**Issue by RainerZufall9393** [calcom/cal.com#7370]
When I use the autofill function for my phone number, it is not inserted correctly. It is a German number, but it is inserted in the wrong format.

https://user-images.githubusercontent.com/126205884/221208300-d23ab075-678c-47e9-b6ee-a0c0b6e31be5.MP4

**Comment by esau-morais** [comment]
I'm not getting to reproduce this issue 😥

**Comment by Ashish0253** [comment]
@RainerZufall9393 can you help me reproduce this error as the language on your screen is not English

**Comment by RainerZufall9393** [comment]
Hey @Ashish0253 how exactly can I help you?

**Comment by Ashish0253** [comment]
Thanks for asking @RainerZufall9393 but I have reproduced this error and I think it's taking the country code of your current location, were you in Germany when you filled in your number ??

**Comment by RainerZufall9393** [comment]
@Ashish0253 
Yes exactly, was in Germany at the time. In fact, the problem is that the country code is not recognized in this example.

At the beginning of my number is +49 which is removed when inserting. That means only the rest of the number is inserted (17672868292), without the +49 at the beginning.

For this reason +1 is recognized as country code by the form, because the number continues after the +49 with a 1.

**Comment by Ashish0253** [comment]
@RainerZufall9393 
Can you please try again to fill your number and confirm whether the error is still there

**Comment by RainerZufall9393** [comment]
@Ashish0253 
Yes, the error is still there

**Comment by ExplorerAadi** [comment]
Yes this has been a common issue with `react-phone-input-2` library that we are using that I have seen in my previous projects too but it's hard to fix this issue since that library does not expose a paste event handler.

Although one way to achieve the desired use case you want @RainerZufall9393  with just a bit of extra effort is to `select all` in the input field before pasting a number with country code already present.
Here is a demo:

https://github.com/calcom/cal.com/assets/52901335/9ceea554-35a6-4835-97f6-f3ddc804b679

**Comment by chengyixu** [comment]
## Proposal

**Problem:** When using browser autofill for phone numbers, international numbers (e.g., German) are inserted in the wrong format. The phone input component does not correctly parse autofilled values with country codes.

**Solution:** Fix the phone number input component to properly handle browser autofill values, correctly detecting the country code and formatting the number.

**Implementation:**
1. Investigate the phone input component (likely using a library like react-phone-number-input or libphonenumber-js)
2. Add an onChange/onInput handler that re-parses autofilled values through the phone number library
3. Ensure country code detection works for autofilled values (E.164 format parsing)
4. Test with various international number formats (DE, UK, US, etc.)

JS/TS/React developer with 30+ published npm tools. Happy to fix this bug.

GitHub: @chengyixu

**Comment by aasthag01-26** [comment]
Hi! I'd like to work on this issue if it's still available. Could someone please assign it to me?
